code/modules/reagents/reagent_dispenser.dm

```diff
@@ -300,25 +300,48 @@
 
 //Beer Kegs
 
-/obj/structure/reagent_dispensers/beerkeg
-	name = "beer keg"
-	desc = "A beer keg"
+/obj/structure/reagent_dispensers/keg
+	name = "keg"
+	desc = "An empty keg."
 	icon_state = "beertankTEMP"
 	amount_per_transfer_from_this = 10
+	var/reagentid = "beer"
+	var/filled = TRUE
 
-/obj/structure/reagent_dispensers/beerkeg/Initialize()
+/obj/structure/reagent_dispensers/keg/Initialize()
+	. = ..()
+	if(filled)
+		reagents.add_reagent(src.reagentid,capacity)
+
+/obj/structure/reagent_dispensers/keg/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/stack/rods))
+		var/obj/item/stack/rods/R = W
+		if(!R.can_use(3)) // like a tripod
+			to_chat(user, span("notice", "You need three rods to make a still!"))
+			return
+		if(do_after(user, 20))
+			if (QDELETED(src))
+				return
+			R.use(3)
+			new /obj/structure/distillery(src.loc)
+			if(reagents)
+				to_chat(user, span("notice", "As you prop the still up on the rods, the reagents inside are spilled. However, you successfully make the still."))
+				reagents.trans_to_turf(get_turf(src), reagents.total_volume)
+			else
+				to_chat(user, span("notice", "You successfully build a still."))
+			qdel(src)
+		return
 	. = ..()
-	reagents.add_reagent("beer",capacity)
 
-/obj/structure/reagent_dispensers/xuizikeg
+/obj/structure/reagent_dispensers/keg/beerkeg

```

Comment:
Why have this if the default is already filled with beer?

Reply:
Because the default is also empty.